src/controllers/sagesseController.ts

```diff
@@ -1,234 +1,270 @@
+
+      let personalWork = descripTableNode!.querySelector('dd')!.querySelector('dl')!.querySelectorAll('dd')[3]!.innerHTML;
+      let CM = hourTableNode!.querySelectorAll('td')[0]!.textContent;
+      let CMTD = hourTableNode!.querySelectorAll('td')[1]!.textContent;
+      let TD = hourTableNode!.querySelectorAll('td')[2]!.textContent;
+      let TP = hourTableNode!.querySelectorAll('td')[3]!.textContent;
+      let terrain = hourTableNode!.querySelectorAll('td')[4]!.textContent;
+      let projet = hourTableNode!.querySelectorAll('td')[5]!.textContent;
+
+      let description = descripTableNode!.children[5]!.textContent!.trim();
+      let context = descripTableNode!.children[7]!.textContent!.trim();
+      let content = descripTableNode!.children[9]!.textContent!.trim();
+
+      if(Array.from(dom.window.document.getElementsByClassName('listeDescendants')).length > 0){    //the module have child (subjects associated to it)
+        let modules = Array.from(dom.window.document.getElementsByClassName('listeDescendants')[0].children[1].children).map(subject => {
+          let subjectSplit = (subject as Element)!.querySelector('p')!.textContent!.split(':');
+          let id = subjectSplit[0].trim();
+          let title = subjectSplit[1].trim();
+          return {'id':id, 'title':title};
+        });
+
+        details = {'personalWork':personalWork ,
+          'CM':CM ,
+          'TD':TD ,
+          'CMTD':CMTD ,
+          'TP':TP ,
+          'Projet':projet ,
+          'Terrain':terrain ,
+          'description':description ,
+          'context':context ,
+          'content':content,
+          'modules':modules
+        };
+
+        res.type('application/json');
+        res.status(200);
+      }
+      else{ //the module don't have child (no subject associated to it)
+        details = {'personalWork':personalWork ,
+          'CM':CM ,
+          'TD':TD ,
+          'CMTD':CMTD ,
+          'TP':TP ,
+          'Projet':projet ,
+          'Terrain':terrain ,
+          'description':description ,
+          'context':context ,
+          'content':content,
+          'modules':[]
+        };
+        res.type('application/json');
+        res.status(200);
+      }
+    } catch(error){
+      console.log('Problem while scrapping HTML (element not found)');
+      details = {'error':'can\'t get elements'};
+      res.status(500);
     }
 
-
-
-
-  })
+    res.send(details); //Send the response
+  });
 };
 
 
-export const getYearDetails = async (yearId: string, year: string, res: any) => {
-  Request.get("https://sagesse.polytech.umontpellier.fr/syllabus/"+yearId+"/"+year, (error: any, response: any, body: any) => {
+export const getYearDetails = async (yearId: string, year: string, res: Response) => {
+  Request.get('https://sagesse.polytech.umontpellier.fr/syllabus/'+yearId+'/'+year, (error: any, response: ResponseReq, body: any) => {
```

Comment:
error and body shoul have a type also

Reply:
I add the type "string" for body, and the type "object" for error